connector/spanmetricsconnector/connector.go

```diff
@@ -289,12 +291,23 @@ func (p *connectorImp) resetState() {
 		p.resourceMetrics.RemoveEvictedItems()
 		p.metricKeyToDimensions.RemoveEvictedItems()
 
-		// Exemplars are only relevant to this batch of traces, so must be cleared within the lock
-		if p.config.Histogram.Disable {
+		// If no histogram and no metrics expiration is configured, we can skip the remaining operations.
+		if p.config.Histogram.Disable && p.config.MetricsExpiration == 0 {
```

Comment:
If a user disables histogram but sets `MetricsExpiration` what should happen? Or vise versa?

Reply:
Then we should do `forEach` for the resource metrics, since either of those features (if enabled) necessitates going through each item and doing an operation on it. If neither of those features are enabled, it't not necessary and we can return early. I'll clarify it a bit in the docs.